File: PatientFinal/PatientFinal/DTOs/ResponseDto.cs
namespace PatientFinal.DTOs;

public class ResponseDto
{
    public string response { get; set; }
    public List<GetAppointmentsDto> Appointments { get; set; } = new List<GetAppointmentsDto>();
    public bool Success { get; set; }
    public static ResponseDto CreateCancellationSuccess()
    {
        return new ResponseDto 
        { 
            response = "Appointment cancelled successfully",
            Success = true 
        };
    }
    public static ResponseDto CreateCancellationError(string errorMessage)
    {
        return new ResponseDto 
        { 
            response = $"Failed to cancel appointment: {errorMessage}",
            Success = false 
        };
    }
    
}

File: PatientsServer/src/main/java/DTOs/ResponseDto.java
package DTOs;

import java.util.ArrayList;
import java.util.List;

public class ResponseDto
{
  private String response;
  private List<GetAppointmentsDto> appointments = new ArrayList<>();

  //---for changing status to Cancelled----------------------------------
  private boolean success;
  public ResponseDto() {}

  public ResponseDto(String response, boolean success) {
    this.response = response;
    this.success = success;
  }

  public static ResponseDto createCancellationSuccess() {
    return new ResponseDto("Appointment cancelled successfully", true);
  }

  public static ResponseDto createCancellationError(String errorMessage) {
    return new ResponseDto("Failed to cancel appointment: " + errorMessage, false);
  }

  public boolean isSuccess() {
    return success;
  }

  public void setSuccess(boolean success) {
    this.success = success;
  }
//----------------------------------------------------------------



  public String getResponse() {
    return response;
  }


  public void setResponse(String response) {
    this.response = response;
  }

  public List<GetAppointmentsDto> getAppointments() {
    return appointments;
  }

  public void setAppointments(List<GetAppointmentsDto> appointments) {
    this.appointments = appointments;
  }
}
